src/main/java/org/jenkinsci/test/acceptance/plugins/git/GitScm.java

```diff
@@ -29,13 +29,21 @@
 import org.openqa.selenium.support.ui.Select;
 
 import javax.swing.text.html.HTMLDocument;
+import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
 
 @Describable("Git")
 public class GitScm extends Scm {
+
     private final Control branch = control("branches/name");
     private final Control url = control("userRemoteConfigs/url");
     private final Control tool = control("gitTool");
+    private final Control repositoryBrowser = this.control(new String[]{"/"});
+    private final Control urlRepositoryBrowser = this.control(new String[]{"browser/repoUrl"});
+    // UI components for additional configuration for some repository browsers
+    private final Control projectName = this.control(new String[]{"browser/projectName"}); // gitlib & viewgit
```

Comment:
These should be factored out to SCM browser abstraction so `GitScm` API si not polluted by gitlab specific controls.

Reply:
I have refactor these controls and added new sub-classes to wrap this specifics UI components.